apps/nsqd/main.go

```diff
@@ -0,0 +1,100 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"math/rand"
+	"os"
+	"path/filepath"
+	"sync"
+	"syscall"
+	"time"
+
+	"github.com/BurntSushi/toml"
+	"github.com/judwhite/go-svc/svc"
+	"github.com/mreiferson/go-options"
+	"github.com/nsqio/nsq/internal/lg"
+	"github.com/nsqio/nsq/internal/version"
+	"github.com/nsqio/nsq/nsqd"
+)
+
+type program struct {
+	once sync.Once
+	nsqd *nsqd.NSQD
+}
+
+func main() {
+	prg := &program{}
+	if err := svc.Run(prg, syscall.SIGINT, syscall.SIGTERM); err != nil {
+		logFatal("%s", err)
+	}
+}
+
+func (p *program) Init(env svc.Environment) error {
+	if env.IsWindowsService() {
+		dir := filepath.Dir(os.Args[0])
+		return os.Chdir(dir)
+	}
```

Comment:
@mreiferson It doesn't hurt anything to have this, but the package does this bit for you now. I generally consider it impolite to `os.Chdir` but in this case I thought it was warranted. https://github.com/judwhite/go-svc/blame/d83e900e4a688aad49f41263bebf8793f0bf6add/svc/svc_windows.go#L59-L66